src/ray/rpc/gcs_server/gcs_rpc_client.h

```diff
@@ -97,8 +97,9 @@ class Executor {
                 : Status(StatusCode(reply.status().code()), reply.status().message()); \
         callback(status, reply);                                                       \
         delete executor;                                                               \
+      } else if (gcs_is_down_) {                                                       \
+        queued_executors_.emplace_back(executor);                                      \
```

Comment:
should be bound the `queued_executors_` so it doesn't oom?

Reply:
I had the same concern here. Then I was thinking about two options:

1. fail the request when it reaches the limit: it might crash raylet as well.
2. retry it immediately, it might OOM as well

Now I think, another way is to block the thread until GCS is back. Let me investigate it.
I'll update the description to include this one.